### DISCUSSION TITLE
HMR not working sometimes  on webpack 5.74.0

### DISCUSSION THREAD

**Discussion by tiger8888** [webpack/webpack#16288] [Q&A] [answered]
What is the current behavior?
Hot module reloading is not  refresh page show after change sub component in vue.

If the current behavior is a bug, please provide the steps to reproduce.

Clone https://github.com/tiger8888/webpack-dev-server-bug.
Run npm install.
Run npm run dev.
then:
1. change `<div>sub</div>` to `<div>sub111</div>` in `Sub.vue`
2. change `console.log("111")` to `console.log("111222")` in `App.vue`
3. change `<div>sub111</div>` to `<div>sub111222</div>` in `Sub.vue`
then the page show is still sub111, and the page is not change after the 3th step's change.   see the pic in `ReadMe.md`

What is the expected behavior?
When making a change to`Sub.vue`, the page is corresponding change the show
Other relevant information:

webpack version:  5.74.0
Node.js version: v14.16.1
Operating System: MacOS
Additional tools

Let me know if you need any more info!  

Thanks ahead!

**Comment by alexander-akait** [comment, marked as answer]
Please use `eval-cheap-source-map` for `devtool`

**Comment by alexander-akait** [comment]
Bug in `vue-loader`, because if you put `console.log("test');` in `Sub.vue` all will work fine, maybe updating the loader and plugin to the latest stable version fix it.

The `script` and `styles` tags work, but the `template` is buggy, we can't fix it here, sorry

**Comment by alexander-akait** [comment]
Feel free to questions and feedback

**Comment by tiger8888** [comment]
but it works well in webpack4

**Reply by alexander-akait** [reply]
I think the problem somewhere in `vue-loader`, because webpack doesn't create magic here, also it works fine with react/preact/angular/etc, maybe some optimizations on vue-loader side break it...
